### PR TITLE
Refactor spawner to be able to reuse code for ros2controlcli

### PR DESCRIPTION
**Pull request by saikishor**
@christophfroehlich has mentioned that ros2 control CLI is missing the functionality to load controllers with the parameter files as we have now with the spawner. upon reviewing the code, the code would need some refactoring to reduce some duplication of code in both places and to be able also to load controllers with namespaced controller manager

This PR would be the first part of refactoring that can go along with #1562, once both the PRs are done, the ros2 control CLI can be modified to add the parameter parsing from the file and proper usage with namespacing.

### CODE REVIEW

**Review by christophfroehlich — APPROVED** [review]
Thanks for addressing this, this PR lgtm.

File: controller_manager/controller_manager/controller_manager_services.py [hunk 1]
```diff
@@ -26,6 +26,29 @@
...
+class bcolors:
+    HEADER = "\033[95m"
+    OKBLUE = "\033[94m"
+    OKCYAN = "\033[96m"
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    ENDC = "\033[0m"
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
```
Comment (fmauch) [inline controller_manager/controller_manager/controller_manager_services.py:51]: Please see this only as a comment, not as a request:

This might be considered to be moved to a different file? Importing this from the controller_manager_services module seems to work fine for the given use cases, but maybe it would be more intuitive to declare it in the __init__ file directly or move it to a separate helpers module.

**Review by fmauch — APPROVED** [review]
This looks like some nice refactoring.

File: controller_manager/controller_manager/controller_manager_services.py (hunk 1 above)
Comment (saikishor) [reply controller_manager/controller_manager/controller_manager_services.py:51]: At the very beginning i wanted to do it in the __init__ file itself, but then if that's the case I couldn't use it inside the controller_manager_services file and then later add it as a module inside the __init__ file. So, in theory you should be able to import in other python files as `from controller_manager import bcolors`

Comment (destogl) [reply controller_manager/controller_manager/controller_manager_services.py]: And besides that, we should rethink on renaming colors, the names are quite fuzzy.

**Review by destogl — APPROVED** [review]
Do you think we should backport this?

I think it is worth the work as it makes user-interface nicer and make it easier to maintain for few more `humble` year.s

File: controller_manager/controller_manager/controller_manager_services.py (hunk 1 above)
Comment (saikishor) [reply controller_manager/controller_manager/controller_manager_services.py]: Sure! Good idea. Maybe we can do it in a separate PR? What do you say?

**Comment by saikishor** [comment]
Sure! I agree it's much easier for maintenance

File: controller_manager/controller_manager/spawner.py [hunk 2]
```diff
@@ -162,14 +122,6 @@ def main(args=None):
...
         required=False,
     )
-    parser.add_argument(
-        "--fallback_controllers",
-        help="Fallback controllers list are activated as a fallback strategy when the"
-        " spawned controllers fail. When the argument is provided, it takes precedence over"
-        " the fallback_controllers list in the param file",
-        default=None,
-        nargs="+",
-    )
```
Comment (bmagyar) [inline controller_manager/controller_manager/spawner.py:172]: This is supposed to be a refactor. Why are we removing things without deprecation? Should go at least into a separate PR so it shows in the changelogs

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: controller_manager/controller_manager/spawner.py (hunk 2 above)
Comment (saikishor) [reply controller_manager/controller_manager/spawner.py:172]: You are right! Sorry my bad. 
The reason I removed it is that the feature of fallback controllers is not complete, so it's better to remove it than add a deprecation note at this point. 

What do you think?

Comment (destogl) [reply controller_manager/controller_manager/spawner.py:172]: I agree with you @saikishor.

This feature is not know very well as it is new and probably nobody exept yourself is using it right now.

Comment (bmagyar) [reply controller_manager/controller_manager/spawner.py:172]: thank you, let's roll with it then!

File: controller_manager/controller_manager/controller_manager_services.py (hunk 1 above)
Comment (bmagyar) [reply controller_manager/controller_manager/controller_manager_services.py]: renaming separately

**Review by bmagyar — APPROVED** [review]
